src/main/drivers/accgyro/accgyro_imuf9001.c

```diff
@@ -0,0 +1,351 @@
+        case GTBCM_GYRO_ACC_PASSTHRU:
+        case GTBCM_GYRO_ONLY_FILTER_F:
+        case GTBCM_GYRO_ACC_FILTER_F:
+        case GTBCM_GYRO_ACC_QUAT_FILTER_F:
+            return (gyroToBoardCommMode_t)commMode;
+            break;
+        default:
+            return GTBCM_DEFAULT;
+    }
+}
+
+uint16_t imufGyroAlignment(void)
+{
+    if (isBoardAlignmentStandard(boardAlignment()))
+    {
+        if(gyroConfig()->gyro_align <= 1)
+        {
+            return 0;
+        }
+        else
+        {
+            return (uint16_t)(gyroConfig()->gyro_align - 1);
+        }
+    }
+    else
+    {
+        return (uint16_t)IMU_CW0;
+    }
+}
+
+void imufSpiGyroInit(gyroDev_t *gyro)
+{
+    uint32_t attempt = 0;
+    imufCommand_t txData;
+    imufCommand_t rxData;
+
+    rxData.param1 = VerifyAllowedCommMode(gyroConfig()->imuf_mode);
+    rxData.param2 = ( (uint16_t)(gyroConfig()->imuf_rate+1) << 16)              | (uint16_t)gyroConfig()->imuf_w;
+    rxData.param3 = ( (uint16_t)gyroConfig()->imuf_roll_q << 16)              | (uint16_t)gyroConfig()->imuf_pitch_q;
+    rxData.param4 = ( (uint16_t)gyroConfig()->imuf_yaw_q << 16)               | (uint16_t)gyroConfig()->imuf_roll_lpf_cutoff_hz;
+    rxData.param5 = ( (uint16_t)gyroConfig()->imuf_pitch_lpf_cutoff_hz << 16) | (uint16_t)gyroConfig()->imuf_yaw_lpf_cutoff_hz;
+    rxData.param6 = ( (uint16_t)0 << 16)                                      | (uint16_t)0;
+    rxData.param7 = ( (uint16_t)0 << 16)                                      | (uint16_t)0;
+    rxData.param8 = ( (int16_t)boardAlignment()->rollDegrees << 16 )          | imufGyroAlignment();
+    rxData.param9 = ( (int16_t)boardAlignment()->yawDegrees << 16 )           | (int16_t)boardAlignment()->pitchDegrees;
+
+    for (attempt = 0; attempt < 10; attempt++)
+    {
+        if(attempt)
+        {
+            resetImuf9001();
+            delay(300 * attempt);
+        }
+
+        if (imuf9001SendReceiveCommand(gyro, IMUF_COMMAND_SETUP, &txData, &rxData))
+        {
+            //enable EXTI
+            mpuGyroInit(gyro);
+            return;
+        }
+    }
+    setArmingDisabled(ARMING_DISABLED_NO_GYRO);
+}
+
+bool imufReadAccData(accDev_t *acc) {
+    UNUSED(acc);
+    return true;
+}
+
+bool imufSpiAccDetect(accDev_t *acc)
+{
+    acc->initFn = imufSpiAccInit;
+    acc->readFn = imufReadAccData;
+
+    return true;
+}
+
+bool imufSpiGyroDetect(gyroDev_t *gyro)
+{
+    // MPU6500 is used as a equivalent of other gyros by some flight controllers
+    switch (gyro->mpuDetectionResult.sensor) {
+    case IMUF_9001_SPI:
+        break;
+    default:
+        return false;
+    }
+
+    gyro->initFn = imufSpiGyroInit;
+    gyro->scale = 1.0f;
+    gyro->mpuConfiguration.resetFn = resetImuf9001;
+    return true;
+}
+
+void imufStartCalibration(void)
+{
+    isImufCalibrating = IMUF_IS_CALIBRATING; //reset by EXTI
+}
+
+void imufEndCalibration(void)
+{
+    isImufCalibrating = IMUF_NOT_CALIBRATING; //reset by EXTI
+}
```

Comment:
No newline at end of file

Reply:
Classy as always, Andrey.
![image](https://user-images.githubusercontent.com/37648729/41513158-6d206cf8-724c-11e8-8883-8c4ca50b1180.png)

Reply:
@piiigletTheThird: Please refrain from posting unrelated content here.